### PR TITLE
Implement persistent update type

### PR DESCRIPTION
**Pull request by Pessimistress**
- new update type: persistent (no time window, never pruned)
- `null` clears a persistent stream

### CODE REVIEW

**Comment by jlisee** [comment]
So I really like this approach.  What would an XVIZ message look that generates one of these null updates?

**Comment by Pessimistress** [comment]
> What would an XVIZ message look that generates one of these null updates?

We have not talked about this. The same goes for INCREMENTAL updates. As of today the parser never generates `null` for any stream.

We may add a field to updates? i.e.

```
{
  update_type: <type>,
  updates: {
    delete: <streamNames>
  }
}
```

**Comment by twojtasz** [comment]
What does this have to do with the parser?  Isn't this just semantics on the message itself.  Really between the message and the semantics of **XVIZStreamBuffer**.

Oh ... the **XVIZStreamBuffer** operates on the objects from the parser.   So, grr, this is curious.

Ok, so I recall previously that `null` was problematic for other reasons, and Nick Chapman had output an empty array.  Looking through the JS code I see that we will validate the primitives and if they are not valid, we don't really add them to the resulting.

Given that, adding a field `delete` doesn't really change what has to happen in the parser.  It semantically the same signal just more explicit.  If we did go with a field i'd prefer **empty** over **delete**, but i'm not feeling that approach just yet.

File: modules/parser/src/synchronizers/xviz-stream-buffer.js [hunk 1]
```diff
@@ -335,6 +348,31 @@ export default class XVIZStreamBuffer {
   }
   /* eslint-enable complexity, no-unused-expressions */
 
+  _mergePersistentSlice({streams}) {
+    const oldStreams = (this.persistent && this.persistent.streams) || {};
+    const newStreams = Object.assign({}, oldStreams);
+
+    for (const streamName in streams) {
+      assert(
```
Comment (twojtasz) [inline modules/parser/src/synchronizers/xviz-stream-buffer.js]: Does this get removed in production code? Still not sure how if this is a soft error or hard error.

**Comment by twojtasz** [comment]
It is very nice that main point of semantics is simply in the streambuffer.

File: modules/parser/src/synchronizers/xviz-stream-buffer.js (hunk 1 above)
Comment (Pessimistress) [reply modules/parser/src/synchronizers/xviz-stream-buffer.js]: Technically the collision doesn't break anything, just the non-persistent updates will be ignored. We can move it into the builder?

**Comment by Pessimistress** [comment]
> I recall previously that null was problematic for other reasons

If that is the case, the parser needs to check if a stream update is empty and convert it to a `null`. Empty arrays will mess up the synchronizers.

**Comment by twojtasz** [comment]
Based on the protobuf work, where there are some changes required due to the way protobufjs instances are structure, we will likely need to settle on `null` as the indication for no-data as the situations where an empty array can be present happens frequently in the protobuf case due to the library we are using.

**Pull request merged**